SetupDataPkg/Tools/KnobService.py

```diff
@@ -127,13 +127,133 @@ def get_assert_style(uefi, assert_string, msg):
     return "C_ASSERT" + assert_string + ");"
 
 
+# put a SPDX header on the generated files
+def get_spdx_header(header, efi_type):
+    spdx = "/** @file {}".format(header)
+    spdx += get_line_ending(efi_type) + get_line_ending(efi_type)
+    spdx += get_spacing_string(efi_type)
+    spdx += "Autogenerated configuration headers from mu_feature_config"
+    spdx += get_line_ending(efi_type) + get_line_ending(efi_type)
+    spdx += get_spacing_string(efi_type)
+    spdx += "Copyright (c) Microsoft Corporation."
+    spdx += get_line_ending(efi_type)
+    spdx += get_spacing_string(efi_type)
+    spdx += "SPDX-License-Identifier: BSD-2-Clause-Patent"
+    spdx += get_line_ending(efi_type)
+    spdx += get_line_ending(efi_type)
+    spdx += "**/"
+    spdx += get_line_ending(efi_type)
+    spdx += get_line_ending(efi_type)
+    return spdx
+
+
+# for UEFI, calculate what variable list size is for a given knob
+def get_variable_list_size(knob):
+    # fixed size per knob is Guid, Attributes, CRC, NameSize, DataSize
+    # we have to calculate the c sizes, not the bloated python sizes
+    # hence why __sizeof__() is not used here
+    size = 16 + 4 + 4 + 4 + 4
+
+    # variable size per knob is len of data and name
+    # the name we convert from schema ASCII to UEFI unicode
+    size += (len(knob.name) + 1) * 2

```

Comment:
nit: adjust comment to indicate +1 is to account for null terminator

Reply:
Updated. Thanks.